clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperation.groovy

```diff
@@ -0,0 +1,450 @@
+          // We are "updating" a Http proxy to an Https proxy, since the only thing we check
+          // for differences is the certificate.
+          targetProxyName = "$httpLoadBalancerName-$TARGET_HTTPS_PROXY_NAME_PREFIX"
+          task.updateStatus BASE_PHASE, "Creating target proxy $targetProxyName..."
+          targetProxy = new TargetHttpsProxy(
+            name: targetProxyName,
+            sslCertificates: [GCEUtil.buildCertificateUrl(project, httpLoadBalancer.certificate)],
+            urlMap: urlMapUrl,
+          )
+          insertTargetProxyOperation = compute.targetHttpsProxies().insert(project, targetProxy).execute()
+          targetProxyUrl = insertTargetProxyOperation.getTargetLink()
+
+          // We should also delete the old Http proxy.
+          deleteExistingProxy = true
+          break
+        case "targetHttpsProxies":
+          targetProxyName = "$httpLoadBalancerName-$TARGET_HTTPS_PROXY_NAME_PREFIX"
+          task.updateStatus BASE_PHASE, "Updating target proxy $targetProxyName..."
+          TargetHttpsProxiesSetSslCertificatesRequest setSslReq = new TargetHttpsProxiesSetSslCertificatesRequest(
+            sslCertificates: [GCEUtil.buildCertificateUrl(project, httpLoadBalancer.certificate)],
+          )
+          compute.targetHttpsProxies().setSslCertificates(project, targetProxyName, setSslReq).execute()
+          UrlMapReference urlMapRef = new UrlMapReference(urlMap: urlMapUrl)
+          def setUrlMapOp = compute.targetHttpsProxies().setUrlMap(project, targetProxyName, urlMapRef).execute()
+          targetProxyUrl = setUrlMapOp.getTargetLink()
+          break
+        default:
+          throw new GoogleHostRule("Updating Http load balancer $httpLoadBalancerName failed. " +
+            "Could not update target proxy $targetProxyName; Illegal target proxy type.")
+          break
+      }
+    } else {
+      targetProxyUrl = existingProxy.getSelfLink()
+    }
+
+    // ForwardingRule
+    if (!forwardingRuleExists) {
+      task.updateStatus BASE_PHASE, "Creating global forwarding rule $httpLoadBalancerName..."
+      def forwardingRule = new ForwardingRule(
+        name: httpLoadBalancerName,
+        iPAddress: httpLoadBalancer.ipAddress,
+        iPProtocol: httpLoadBalancer.ipProtocol,
+        portRange: httpLoadBalancer.certificate ? "443" : httpLoadBalancer.portRange,
+        target: targetProxyUrl,
+      )
+      compute.globalForwardingRules().insert(project, forwardingRule).execute()
+    } else if (forwardingRuleExists && forwardingRuleNeedsUpdated) {
+      task.updateStatus BASE_PHASE, "Updating global forwarding rule $httpLoadBalancerName..."
+      def forwardingRule = new ForwardingRule(
+        name: httpLoadBalancerName,
+        iPAddress: httpLoadBalancer.ipAddress ?: existingRule.getIPAddress(),
+        iPProtocol: httpLoadBalancer.ipProtocol ?: existingRule.getIPProtocol(),
+        portRange: httpLoadBalancer.certificate ? "443" : httpLoadBalancer.portRange,
```

Comment:
Does it have to be port `443`, or can it be user specified? I realize 443 is a good default